src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -286,11 +286,17 @@
         "JARS?", KeywordSegment, name="jar", type="file_keyword"
     ),
     NoscanKeywordSegment=StringParser(
-        "NOSCAN", SymbolSegment, name="noscan_keyword", type="keyword"
+        "NOSCAN", KeywordSegment, name="noscan_keyword", type="keyword"
     ),
     WhlKeywordSegment=StringParser(
         "WHL", KeywordSegment, name="whl", type="file_keyword"
     ),
+    SQLConfPropertiesSegment=StringParser(
+        "-v",
+        SymbolSegment,
+        name="sql_conf_properties_parameter",
+        type="sql_conf_properties",
+    ),
```

Comment:
I defined this segment to match "-v" thinking I would get similar behavior to `EqualsSegment_a` which matches on "=="; however, the parser is still splitting "-v" into "-" and "v". I've tried a test case where "-v" is isolated w/o any`SET` statement and added `SQLConfPropertiesSegment`to `StatementSegment` when doing this with no luck.

```bash
INFO       Found unparsable segment...
INFO       [L:  3, P:  4]      |unparsable:                                                   !! Expected: 'Nothing...'
[L:  3, P:  4]      |    whitespace:                                               ' '
[L:  3, P:  5]      |    raw:                                                      '-'
[L:  3, P:  6]      |    raw:                                                      'v'
```

Any suggestions on how to match/parse this? Using a combination of `MinusSegment` and `SingleIdentifierSegment` works but tries to enforce a whitespace after the "-" sign. Seemed like creating a new SymbolSegment would be a better solution than modifying a rule for this one edge-case for the spark dialect.

Reply:
Got it running 🙂  just needed to insert a new `StringLexer` before "minus".

Will clean up then remove DRAFT status.

Reply:
Got it running 🙂  just needed to insert a new `StringLexer` before "minus".

Will clean up then remove DRAFT status.

Reply:
Hmmm... I'm wondering if this si the right way of going about it?

Let's say you had an identifier called `-vmycolumn`. That would now be lexed and parsed as two identifiers.

That might be pretty unlikely but what about `SELECT col1-variable1`?

I think for the one exception it might be better to define `SQLConfPropertiesSegment` as being made up of two symbols:

```
    SQLConfPropertiesSegment=Sequence(
        Ref("MinusSegment"),
        StringParser("v", SymbolSegment, name="sql_conf_option_set", type="sql_conf_option"),
        allow_gaps=False,
    ),
```

Reply:
I tried a similar approach but kept getting the below:

```bash
INFO       LINTING (test/fixtures/dialects/spark3/set.sql)
== [test/fixtures/dialects/spark3/set.sql] LINTING
== [test/fixtures/dialects/spark3/set.sql] FAIL
L:   3 | P:   5 | L006 | Missing whitespace after -
==== summary ====
files:             1 violations:        1
clean files:       0 unclean files:     1
avg per file:   1.00 unclean rate:   100%
status:         FAIL
All Finished 📜 🎉!
```

Linter keeps suggesting a whitespace even though `allow_gaps` is False. Based on your feedback, seems like an exception to L006 is simpler than defining a new `StringLexer`

Reply:
Ah this works better then:

```
    SQLConfPropertiesSegment=Sequence(
        StringParser("-", SymbolSegment, name="dash", type="dash"),
        StringParser("v", SymbolSegment, name="sql_conf_option_set", type="sql_conf_option"),
    ),
```

Reply:
And this has just helped me fix #2771!